python/cuml/test/test_linear_model.py

```diff
@@ -347,6 +349,36 @@ def test_logistic_regression_model_default(dtype):
     assert culog.score(X_test, y_test) >= sklog.score(X_test, y_test) - 0.022
 
 
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("order", ["C", "F"])
+@pytest.mark.parametrize("sparse_input", [False, True])
+@pytest.mark.parametrize("fit_intercept", [False, True])
+@pytest.mark.parametrize("penalty", ["none", "l1", "l2"])
+def test_logistic_regression_model_digits(
+        dtype, order, sparse_input, fit_intercept, penalty):
+
+    # smallest sklearn score
+    # put it as a constant here, because sklearn 0.23.1 needs a lot of iters
+    # to converge and has a bug returning an unrelated error if not converged.
+    acceptable_score = 0.95
+
+    digits = load_digits()
```

Comment:
Thanks for pointing me to this dataset! I've added a basic test with this dataset and updated the existing decision/predict tests to work also on sparse data.

I don't worry about the behavior of this solver on sparse data much, because (1) the code paths for dense/sparse variant differ only in one step (matrix multiplication features*weights), and (2) the optimization space (model weights) is dense in both cases anyway.

Reply:
> the code paths for dense/sparse variant differ only in one step

Indeed this particular detail made your changes much easier to review. 

Thank you for adding this dataset to the tests! I've found it useful to verify against data that has realistic sparsity patterns because it's exposed potential numerical issues that led to unexpected results (which might not have presented themselves otherwise). Especially as this code grows over time, having such a test will help  us maintain that piece of mind.